pkg/chunk/cached_store.go

```diff
@@ -742,6 +749,14 @@ func NewCachedStore(storage object.ObjectStorage, config Config) ChunkStore {
 			return float64(used)
 		}))
 	go store.uploadStaging()
+	if store.conf.CacheDir != "memory" {
+		go func() {
+			for {
+				store.cleanDelayStaging()
+				time.Sleep(time.Second)
```

Comment:
1s is too short if there are many pending keys, in which case the **scan** step of `cleanDelayStaging` may cost considerable time & resource